agir/notifications/actions.py

```diff
@@ -2,59 +2,75 @@
-
-    for activity_type in Subscription.DEFAULT_PERSON_EMAIL_TYPES:
-        subscriptions += [
-            Subscription(
-                person=person,
-                type=Subscription.SUBSCRIPTION_EMAIL,
-                activity_type=activity_type,
-            ),
-            Subscription(
-                person=person,
-                type=Subscription.SUBSCRIPTION_PUSH,
-                activity_type=activity_type,
-            ),
-        ]
-
-    return subscriptions
-
-
-def get_default_group_subscriptions(person, membership):
-    subscriptions = []
-
-    for activity_type in Subscription.DEFAULT_GROUP_EMAIL_TYPES:
-        subscriptions += [
-            Subscription(
-                person=person,
-                type=Subscription.SUBSCRIPTION_EMAIL,
-                activity_type=activity_type,
-                membership=membership,
-            ),
-            Subscription(
-                person=person,
-                type=Subscription.SUBSCRIPTION_PUSH,
-                activity_type=activity_type,
-                membership=membership,
-            ),
-        ]
-
-    return subscriptions
-
-
-def create_default_person_subscriptions(person):
-    subscriptions = get_default_person_subscriptions(person)
+def get_default_person_email_subscriptions(person):
+    return [
+        Subscription(
+            person=person,
+            type=Subscription.SUBSCRIPTION_EMAIL,
+            activity_type=activity_type,
+        )
+        for activity_type in Subscription.DEFAULT_PERSON_EMAIL_TYPES
+    ]
+
+
+def get_default_group_email_subscriptions(person, membership):
+    return [
+        Subscription(
+            person=person,
+            type=Subscription.SUBSCRIPTION_EMAIL,
+            activity_type=activity_type,
+            membership=membership,
+        )
+        for activity_type in Subscription.DEFAULT_PERSON_EMAIL_TYPES
+    ]
+
+
+def get_default_person_push_subscriptions(person):
+    return [
+        Subscription(
+            person=person,
+            type=Subscription.SUBSCRIPTION_PUSH,
+            activity_type=activity_type,
+        )
+        for activity_type in Subscription.DEFAULT_PERSON_PUSH_TYPES
+    ]
+
+
+def get_default_group_push_subscriptions(person, membership):
+    return [
+        Subscription(
+            person=person,
+            type=Subscription.SUBSCRIPTION_PUSH,
+            activity_type=activity_type,
+            membership=membership,
+        )
+        for activity_type in Subscription.DEFAULT_PERSON_PUSH_TYPES
+    ]
+
+
+def create_default_person_email_subscriptions(person):
+    subscriptions = get_default_person_email_subscriptions(person)
+    person_memberships = Membership.objects.filter(person=person).distinct()
+
+    if person_memberships.exists():
+        for membership in person_memberships:
+            subscriptions += get_default_group_email_subscriptions(person, membership)
+
+    Subscription.objects.bulk_create(subscriptions)
```

Comment:
Question, il n'y a pas besoin d'ignorer les conflits sur ces `bulk_create` ?
```suggestion
    Subscription.objects.bulk_create(subscriptions, ignore_conflicts=True)
```

Reply:
Sachant que la fonction n'est appelée que si la personne n'a aucune souscription, je pense pas ?